### PR TITLE
scheduler: limit retry limit for balance-*-scheduler when placement rule is enabled.

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?

ref https://github.com/tikv/pd/issues/2783

When placement rule is enabled, FitRegion will be called during balance-*-scheduler and the numbers of calling `FitRegion` function would be `maxScheduleRetries` * `store count` * `retry limit` * const number(about 3). Then FitRegion would be consumed too much cpu resources which affects overall performance.

### What is changed and how it works?

Add new property `fitRegionCallingCount` in scheduler, when placement rule is enabled, the fitRegionCallingCount is limited by  the fitRegionCallingLimit.

### Check List

Tests

- Unit test

Related changes

- Need to cherry-pick to the release branch

### CODE REVIEW

File: server/schedulers/balance_leader.go [hunk 1]
```diff
@@ -35,6 +35,10 @@ const (
 	BalanceLeaderType = "balance-leader"
 	// balanceLeaderRetryLimit is the limit to retry schedule for selected source store and target store.
 	balanceLeaderRetryLimit = 10
+	// fitRegionCalling is the limit of the count of calling FitRegion in one scheduler Schedule
+	// when placement rule is enabled.
+	// TODO: set a proper value
+	fitRegionCallingLimit = 10
```
Comment (nolouch) [inline server/schedulers/balance_leader.go:41]: The value may cause us cannot find a suitable store.

Comment (disksing) [reply server/schedulers/balance_leader.go:41]: right.

Comment (lhy1024) [reply server/schedulers/balance_leader.go:41]: Should it be proportional to the number of stores

**Comment by lhy1024** [comment]
why close it?

**Pull request closed without merging**